pymc_marketing/customer_choice/nested_logit.py

```diff
@@ -0,0 +1,1064 @@
+        conditional probabilities (`P_y_given`) are computed in the `make_exp_nest`
+        method.
+
+        This is used within the PyMC model to construct the **tree-based aggregation**
+        of utilities, where lower-level nodes (alternative-specific utilities) are
+        passed upward in the tree structure to compute nest-level and top-level
+        choice probabilities.
+
+        Parameters
+        ----------
+        U : TensorVariable
+            Tensor of systematic utilities with shape `(n_obs, n_alternatives)`.
+        W : TensorVariable | None
+            Fixed covariates design matrix (if used), else `None`.
+        betas_fixed_ : TensorVariable
+            Alternative-specific coefficients for the fixed covariates.
+        lambdas_nests : TensorVariable
+            A Beta random variable for each of the nests
+        level : str
+            Which nesting level to compute ("top" or "mid"), indicating whether
+            to compute probabilities for top-level or mid-level nests.
+
+        Returns
+        -------
+        conditional_probs : dict[str, dict[str, TensorVariable]]
+            Dictionary for each nest containing:
+            - "exp": the exponentiated inclusive value of the nest.
+            - "P_y_given": conditional choice probabilities within the nest.
+        nest_probs : dict[str, TensorVariable]
+            Dictionary mapping each nest to its overall selection probability.
+
+        Raises
+        ------
+        ValueError
+            If the nesting structure is invalid or the nest name is not found.
+        """
+        nest_indices = self.nest_indices
+        conditional_probs = {}
+        ## Collect All Exp Inclusive Value terms per nest
+        for n in nest_indices[level].keys():
+            exp_W_nest, P_y_given_nest = self.make_exp_nest(
+                U, W, betas_fixed_, lambdas_nests, n, level
+            )
+            exp_W_nest = pm.math.sum(exp_W_nest, axis=1)
+            conditional_probs[n] = {"exp": exp_W_nest, "P_y_given": P_y_given_nest}
+
+        ## Sum the exp inclusive value terms as normalising constanT
+        denom = pm.Deterministic(
+            f"denom_{level}",
+            pm.math.sum(
+                [conditional_probs[n]["exp"] for n in nest_indices[level].keys()],
+                axis=0,
+            ),
+        )
+        ## Calculate the nest probability
+        nest_probs = {}
+        for n in nest_indices[level].keys():
+            P_nest = pm.Deterministic(f"P_{n}", (conditional_probs[n]["exp"] / denom))
+            nest_probs[n] = P_nest
+        return conditional_probs, nest_probs
+
+    def make_model(self, X, W, y) -> pm.Model:
+        """Build Model."""
+        nest_indices = self.nest_indices
+        coords = self.coords
+
+        with pm.Model(coords=coords) as model:
+            alphas = pm.Normal("alphas", 0, 1, dims="alts")
+            betas = pm.Normal("betas", 0, 1, dims=("alt_covariates"))
```

Comment:
Are these priors to expose in the configuration?

Reply:
Good catch, they weren't being passed through. I've amended that now. 
<img width="1039" alt="image" src="https://github.com/user-attachments/assets/c2a56d2f-e502-4ae9-a903-9ef133f55c96" />